metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -431,26 +406,34 @@ func CreateStackdriverSink(uri *url.URL) (core.DataSink, error) {
 	return sink, nil
 }
 
-func (sink *StackdriverSink) preprocessMemoryMetrics(metricSet *core.MetricSet) *core.MetricSet {
-	usage := metricSet.MetricValues[core.MetricMemoryUsage.MetricDescriptor.Name].IntValue
-	workingSet := metricSet.MetricValues[core.MetricMemoryWorkingSet.MetricDescriptor.Name].IntValue
-	bytesUsed := core.MetricValue{
-		IntValue: usage - workingSet,
-	}
-
-	newMetricSet := &core.MetricSet{
-		MetricValues: map[string]core.MetricValue{},
+func parseBoolFlag(opts map[string][]string, name string, targetValue *bool) error {
+	if len(opts[name]) >= 1 {
+		var err error
+		*targetValue, err = strconv.ParseBool(opts[name][0])
+		if err != nil {
+			return fmt.Errorf("%s = %s is not correct boolean value", name, opts[name][0])
+		}
 	}
+	return nil
+}
 
-	newMetricSet.MetricValues["memory/bytes_used"] = bytesUsed
-
-	memoryFaults := metricSet.MetricValues[core.MetricMemoryPageFaults.MetricDescriptor.Name].IntValue
-	majorMemoryFaults := metricSet.MetricValues[core.MetricMemoryMajorPageFaults.MetricDescriptor.Name].IntValue
+func (sink *StackdriverSink) computeDerivedMetrics(metricSet *core.MetricSet) *core.MetricSet {
+	newMetricSet := &core.MetricSet{MetricValues: map[string]core.MetricValue{}}
+	usage, usageOK := metricSet.MetricValues[core.MetricMemoryUsage.MetricDescriptor.Name]
+	workingSet, workingSetOK := metricSet.MetricValues[core.MetricMemoryWorkingSet.MetricDescriptor.Name]
+	memoryFaults, memoryFaultsOK := metricSet.MetricValues[core.MetricMemoryPageFaults.MetricDescriptor.Name]
```

Comment:
Move closer to the place where it's used

Reply:
Done